### PR TITLE
feat(APP-33273): add Mosaic component

### PR DESCRIPTION
**Pull request by habxtech**
See [APP-33273](https://habxfr.atlassian.net/browse/APP-33273): ui-core: Create Mosaic Component in ui-core.

### CODE REVIEW

**Comment by jean9696** [comment]
@hugomalzieux can you uncommit your change in package.json & package-lock.json please ?

File: src/Mosaic/Mosaic.interface.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react'
+
+export interface MosaicProps {
+    items: React.ReactNode[]
```
Comment (jean9696) [inline src/Mosaic/Mosaic.interface.ts]: Why don't you allow to pass them as children ?

File: src/Mosaic/Mosaic.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import { withGrid } from '../_storybook/withGrid'
+
+import { Mosaic, MosaicProps } from '.'
+
+const ItemContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-size: 14px;
+  background: #dcdcdd;
```
Comment (jean9696) [inline src/Mosaic/Mosaic.stories.tsx]: Please use theme color

Comment (jean9696) [reply src/Mosaic/Mosaic.stories.tsx]: I think you can use the `Background` component here

File: src/Mosaic/Mosaic.style.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    grid-auto-flow: column;
+  }
+
+  &[data-spacing='true'] {
+    gap: 4px;
+  }
+`
+
+export const Item = styled.div`
+  > * {
```
Comment (jean9696) [inline src/Mosaic/Mosaic.style.ts]: Why do you need this ?

File: src/Mosaic/Mosaic.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import {
+  GridContainer,
+  MosaicItem,
+  Item,
+  FavouriteLayout,
+  IconContainer,
+} from './Mosaic.style'
+
+export const Mosaic: React.FunctionComponent<MosaicProps> = (props) => {
+  const structure = React.useMemo((): React.ReactNode[] => {
```
Comment (jean9696) [inline src/Mosaic/Mosaic.tsx]: Why do you need a `useMemo` here ?

**Review by jean9696 — CHANGES_REQUESTED** [review]

File: src/Mosaic/Mosaic.interface.ts (hunk 1 above)
Comment (hugomalzieux) [reply src/Mosaic/Mosaic.interface.ts]: Yes you're right

File: src/Mosaic/Mosaic.style.ts (hunk 3 above)
Comment (hugomalzieux) [reply src/Mosaic/Mosaic.style.ts]: I clean this

File: src/Mosaic/Mosaic.tsx (hunk 4 above)
Comment (hugomalzieux) [reply src/Mosaic/Mosaic.tsx]: old usage, useless now :)

File: src/Mosaic/Mosaic.stories.tsx [hunk 5]
```diff
@@ -27,71 +22,79 @@ const GridContainer = styled.div`
...
-const array = new Array(10)
-  .fill(null)
-  .map((a, i) => (
-    <ItemContainer key={`ìtem-container-${i + 1}`}>{i + 1}</ItemContainer>
-  ))
+const array = new Array(10).fill(null).map((a, i) => (
+  <Background
+    key={`ìtem-container-${i + 1}`}
+    backgroundColor={palette.neutralBlackWithIntensityFading[300]}
+    style={{
```
Comment (jean9696) [inline src/Mosaic/Mosaic.stories.tsx]: Please do not use `style`, prefer usage of `styled(Background)`

File: src/Mosaic/Mosaic.tsx [hunk 6]
```diff
@@ -6,74 +6,68 @@ import { MosaicProps } from './Mosaic.interface'
...
-          </IconContainer>
-        </FavouriteLayout>
-      ) : (
-        mosaic
-      )
-    }
+    return columnNumber === 1 ? (
+      <FavouriteLayout>
+        {mosaic}
+        <IconContainer>
```
Comment (jean9696) [inline src/Mosaic/Mosaic.tsx]: I don't think that the star should be in `Mosaic`, I think it should be managed outside of the `Mosaic` Layout

**Review by jean9696 — CHANGES_REQUESTED** [review]

**Pull request merged**